### ISSUE TITLE
Use telegram userid instead of phone number

### ISSUE THREAD

**Issue by Gennaro552** [majn/telegram-purple#512]
Very often you want to keep the phone number hidden on the telegram while this plugin uses it as a user name. The main problem is that if used in conjunction with other plugins the latter could leak the phone number (otr) as the phone number being considered a public identifier they have no problem showing it. I recommend using the telegram userid as the identifier and ask if there is a temporary solution by changing the settings.

**Comment by BenWiederhake** [comment]
I fully agree that a phone number is sensitive information.  However, where exactly do those plugins leak this information?  Why would they just spray your username in the wild?  Also, what do you do when the user changes their name, changes their `@`-handle, or has no `@`-handle in the first place?

If you figure something out: PRs are very welcome!  I think a good place to start is `telegram-purple.c:862`, in function `tgprpl_init` for configuration; `telegram-purple.c:574` in function `tgprpl_login` for different handling, `tgp-request.c:303` in function `request_value` for the new location of the phone number.

Have fun!
